File: NestedWorld/NestedWorld/Monster.m
//
//  Monster.m
//  NestedWorld
//
//  Created by Jean-Antoine Dupont on 08/12/2015.
//  Copyright © 2015 NestedWorld. All rights reserved.
//

#import "Monster.h"

@implementation Monster

- (id) init:(NSString *)monsterName life:(int)hP attack:(int)aP defense:(int)dP
{
    self = [super init];
    
    if (self != nil) {
        name = monsterName;
        
        healthPoint = hP;
        attackPoint = aP;
        defensePoint = dP;
    }
    
    return self;
}

@end

File: nestedworld/nestedworld/Monster.swift
//
//  Monster.swift
//  nestedworld
//
//  Created by Jean-Antoine Dupont on 17/02/2016.
//  Copyright © 2016 NestedWorld. All rights reserved.
//

import Foundation

class Monster
{
    enum TYPE
    {
        case WATER
        case FIRE
        case UNKNOWN
        
        func toString() -> String
        {
            switch self {
            case .FIRE:
                return "fire"
            case .WATER:
                return "water"
            default:
                return "unknown"
            }
        }
    }
    
    var name: String
    
    var attack: UInt
    var defense: UInt
    var healthPoint: UInt
    
    
    // MARK: ...
    
    init?(name: String, attack: UInt, defense: UInt, hp: UInt)
    {
        self.name = name
        
        self.attack = attack
        self.defense = defense
        self.healthPoint = hp
        
        if (self.checkParams() == false) {
            return nil
        }
    }
    
    
    // MARK: Private functions
    
    private func checkParams() -> Bool
    {
        if (self.name.isEmpty == true) {
            return false
        }
        
        return true
    }
}